Makefile

```diff
@@ -1,4 +1,151 @@
+	$(GO) mod download
+
+help: ## Display this help
+	@awk 'BEGIN {FS = ":.*?## "; printf "\n$(KREPO_DESC)\n\nUsage:\n  make \033[36m<cmd>\033[0m\n"} /^[a-zA-Z0-9._-]+:.*?## / {printf "  \033[36m%-20s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST)
+
+build: $(COMMANDS)  ## Build all artifacts
+
+$(COMMANDS): ## Build artifct
+	CGO_ENABLED=1 $(GO) build -ldflags "$(LDFLAGS)" -o $(BIN_OUTPUT_DIR)/$@ ./cmd/$@
+
+deploy: ## Deploy TriggerMesh stack to default Kubernetes cluster using ko
+	CGO_ENABLED=1 $(KO) deploy -f $(BASE_DIR)/config
+
+undeploy: ## Remove TriggerMesh stack from default Kubernetes cluster using ko
+	$(KO) delete -f $(BASE_DIR)/config
+
+release: ## Build release binaries
+	@set -e ; \
+	for bin in $(COMMANDS) ; do \
+		for platform in $(TARGETS); do \
+			GOOS=$${platform%/*} ; \
+			GOARCH=$${platform#*/} ; \
+			RELEASE_BINARY=$$bin-$${GOOS}-$${GOARCH} ; \
+			[ $${GOOS} = "windows" ] && RELEASE_BINARY=$${RELEASE_BINARY}.exe ; \
+			echo "GOOS=$${GOOS} GOARCH=$${GOARCH} $(GO) build -ldflags "$(LDFLAGS)" -o $(DIST_DIR)/$${RELEASE_BINARY}" ./cmd/$$bin ; \
+			GOOS=$${GOOS} GOARCH=$${GOARCH} $(GO) build -ldflags "$(LDFLAGS)" -o $(DIST_DIR)/$${RELEASE_BINARY} ./cmd/$$bin ; \
+		done ; \
+	done
+
+test: install-gotestsum ## Run unit tests
+	@mkdir -p $(TEST_OUTPUT_DIR)
+	$(GOTEST) -p=1 -race -cover -coverprofile=$(TEST_OUTPUT_DIR)/$(KREPO)-c.out $(GOPKGS)
+
+cover: test ## Generate code coverage
+	@mkdir -p $(COVER_OUTPUT_DIR)
+	$(GOTOOL) cover -html=$(TEST_OUTPUT_DIR)/$(KREPO)-c.out -o $(COVER_OUTPUT_DIR)/$(KREPO)-coverage.html
+
+lint: install-golangci-lint ## Lint source files
+	$(GOLINT) $(GOPKGS)
+
+fmt: ## Format source files
+	$(GOFMT) -s -w $(shell $(GO) list -f '{{$$d := .Dir}}{{range .GoFiles}}{{$$d}}/{{.}} {{end}} {{$$d := .Dir}}{{range .TestGoFiles}}{{$$d}}/{{.}} {{end}}' $(GOPKGS))
+
+fmt-test: ## Check source formatting
+	@test -z $(shell $(GOFMT) -l $(shell $(GO) list -f '{{$$d := .Dir}}{{range .GoFiles}}{{$$d}}/{{.}} {{end}} {{$$d := .Dir}}{{range .TestGoFiles}}{{$$d}}/{{.}} {{end}}' $(GOPKGS)))
+
+verify: lint fmt-test ## Verify code base with lint and fmt
+
+IMAGES = $(foreach cmd,$(COMMANDS),$(cmd).image)
+images: $(IMAGES) ## Builds container images
+$(IMAGES): %.image:
+	$(DOCKER) build -t $(IMAGE_REPO)/$* -f ./cmd/$*/Dockerfile .
+
+CLOUDBUILD_TEST = $(foreach cmd,$(COMMANDS),$(cmd).cloudbuild-test)
+cloudbuild-test: $(CLOUDBUILD_TEST) ## Test container image build with Google Cloud Build
+$(CLOUDBUILD_TEST): %.cloudbuild-test:
+
+# NOTE (antoineco): Cloud Build started failing recently with authentication errors when --no-push is specified.
+# Pushing images with the "_" tag is our hack to avoid those errors and ensure the build cache is always updated.
+	gcloud builds submit $(BASE_DIR) --config cloudbuild.yaml --substitutions _CMD=$*,COMMIT_SHA=${IMAGE_SHA},_KANIKO_IMAGE_TAG=_
+
+CLOUDBUILD = $(foreach cmd,$(COMMANDS),$(cmd).cloudbuild)
+cloudbuild: $(CLOUDBUILD) ## Build and publish image to GCR
```

Comment:
the cloudbuild targets need `cloudbuild.yaml` to be present and the accompanying `.gcloudignore` and such

Reply:
The `cloudbuild.yaml` is included with this PR, and IIRC sans the `.gcloudignore`, google will use the `.gitignore` as a fallback. However can add it.

Reply:
I think the problem with using the `.gitignore` file was that there are things we want to keep in Git, but don't want to push to container images (all the source code for example).

Reply:
Fair. Added the `.gcloudignore` to account for this use case.

Reply:
`.gcloudignore` imports rules from `.dockerignore` and adds its own on top of it.
Would you mind also copying over `.dockerignore`? (I know, we ignore a lot of things)

Reply:
`.dockerignore` is already present.

Reply:
Oh... And added by me on top of that 😅  Sorry